### PR TITLE
(Test) Add Tests for batch number selector

### PR DESCRIPTION
**Pull request by ODORA0**
## Requirements
- [ ] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.

### CODE REVIEW

**Comment by jabahum** [comment]
@ODORA0 can we fix the build failure

File: src/stock-operations/batch-no-selector/batch-no-selector.component.tsx [hunk 1]
```diff
@@ -78,36 +71,26 @@ const BatchNoSelector = <T,>(props: BatchNoSelectorProps<T>) => {
...
-            itemToString={(s: StockBatchDTO) =>
-              s?.batchNo ? `${s?.batchNo} | Qty: ${s?.quantity ?? ""}` : ""
-            }
-            placeholder={props.placeholder}
-            invalid={props.invalid}
-            invalidText={props.invalidText}
-            ref={ref}
-          />
-        )}
+      <ComboBox
```
Comment (jabahum) [inline src/stock-operations/batch-no-selector/batch-no-selector.component.tsx:74]: is there any particular reason this needs to change from a controlledDropDown to just a combox ??

**Comment by hadijahkyampeire** [comment]
@ODORA0 please look into the merge conflicts and Jabar's question

File: src/stock-operations/add-stock-operation/stock-items-addition-row.component.tsx [hunk 2]
```diff
@@ -206,8 +206,6 @@ const StockItemsAdditionRow: React.FC<StockItemsAdditionRowProps> = ({
                       }}
                       placeholder={"Filter..."}
                       invalid={!!errors?.stockItems?.[index]?.stockBatchUuid}
```
Comment (hadijahkyampeire) [inline src/stock-operations/add-stock-operation/stock-items-addition-row.component.tsx:208]: why did you remove these?

File: src/stock-operations/batch-no-selector/batch-no-selector.component.tsx (hunk 1 above)
Comment (hadijahkyampeire) [reply src/stock-operations/batch-no-selector/batch-no-selector.component.tsx:74]: @ODORA0 without controller, the field won't be validated or recorded by the zod implementation, please check that. Also this is a reusable component, we should be careful modifying it.

File: src/stock-operations/batch-no-selector/batch-no-selector.test.tsx [hunk 3]
```diff
@@ -1,11 +1,98 @@
...
+        batchNo: 'Batch1',
+        expiration: new Date('2023-12-31'),
+        stockItemUuid: 'item-uuid-1',
+        quantity: '15',
+        voided: false,
+      });
+    });
+  });
+
+});
```
Comment (hadijahkyampeire) [inline src/stock-operations/batch-no-selector/batch-no-selector.test.tsx:98]: New line here

**Review by hadijahkyampeire — COMMENTED** [review]
Thanks @ODORA0 , please check the comments

**Comment by jabahum** [comment]
@ODORA0 i think we can close this since i see u created another ticket with the same issue

**Comment by ODORA0** [comment]
@jabahum yes, closing in favour of https://github.com/openmrs/openmrs-esm-stock-management/pull/232
